lightning/src/chain/keysinterface.rs

```diff
@@ -292,6 +294,26 @@ pub trait ChannelKeys : Send+Clone {
 	/// chosen to forgo their output as dust.
 	fn sign_closing_transaction<T: secp256k1::Signing>(&self, closing_tx: &Transaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
 
+	/// Create a signature for a delayed transaction, spending a local commitment/HTLC output.
+	///
+	/// Output spending is constrained bya OP_CSV and therefore spending must set accurately
+	/// its nSequence to satisfy encumbrance.
+	///
+	/// It may be called multiples time for same output(s) if a fee-bump is needed but there
+	/// is no unfavorable timelock expiration to consider. Therefore it doesn't have to
+	/// be implemented on the same Lightning channel signer and can be on a "cooler" host.
+	///
+	/// Input index to point towards outpoint spent, committed in sig (BIP143).
+	///
+	/// CSV is the relative lock-time value as expected to be part of witnessScript.
+	///
+	/// Amount is value of the output spent by this input, committed in sig (BIP143).
+	///
+	/// Per_commitment_point is the dynamic point corresponding to the channel state
+	/// detected onchain. It has been generated by us and is used to derive our
+	/// delayed_payment_key.
+	fn sign_delayed_transaction<T: secp256k1::Signing + secp256k1::Verification>(&self, spend_tx: &Transaction, input: usize, on_local_tx_csv: u16, amount: u64, per_commitment_point: &PublicKey, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()>;
```

Comment:
Just in general I think this is the wrong way to go about it - removing the secret key is great, but there's no reason to *require* that a ChannelKeys has this function - the user can chose to sign the delayed transaction any way they want and we never actually need it. We should obviously provide a way to get at the relevant keys from InMemoryChannelKeys so that the user can sign correctly, but if its eg a hardware wallet, its not our job to tell them how to implement the API.

Reply:
Yes should add somewhere keys don't have to reside on same device.